vue/components/ui/molecules/search-persistent/save-filter-modal.vue

```diff
@@ -95,25 +121,148 @@ export const SaveFilterModal = {
                 name: this.filterNameData,
                 value: this.searchData,
-                tenancy: tenancy,
+                tenancy: this.getSelectedTenancy(tenancy),
                 context: "" // TODO
             }));
+        },
+        isTenancyEmpty() {
+            return this.tenancyItemsData.length === 1;
+        },
+        isTenancyFormValid() {
+            for (const item of this.tenancyItemsData) {
+                if (
+                    this.isTenancyChoiceSelected(item.value) &&
+                    this.getSelectedTenancy(item.value) === null
+                ) {
+                    return false;
+                }
+            }
+
+            return true;
+        },
+        userTenancy() {
+            return { choices: [], selectedValue: "user", selectVisible: false };
+        },
+        builtTenancies() {
+            const builtTenancies = {};
+
+            this.tenancyItems.forEach(item => {
+                if (item.value === "user") builtTenancies[item.value] = this.userTenancy;
+                else {
+                    builtTenancies[item.value] = {
+                        choices: [],
+                        selectedValue: null,
+                        selectVisible: false
+                    };
+                }
+            });
+
+            return builtTenancies;
         }
     },
-    mounted: async function() {},
+    mounted: async function() {
+        this.tenancies = this.builtTenancies;
+
+        // Getting brands, channels and factories
+        this.tenancies.brand.choices = await this.getBrands();
+        this.tenancies.channel.choices = await this.getChannels();
+        this.tenancies.factory.choices = await this.getFactories();
+
+        this.removeInvalidChoices();
+    },
     methods: {
+        async getBrands() {
+            // TODO fix me, I'm hardcoded and have 7 options
+            return [
+                { value: "brand_a", label: "Brand A" },
+                { value: "brand_b", label: "Brand B" },
+                { value: "brand_c", label: "Brand C" },
+                { value: "brand_d", label: "Brand D" },
+                { value: "brand_e", label: "Brand E" },
+                { value: "brand_f", label: "Brand F" },
+                { value: "brand_g", label: "Brand G" }
+            ];
+        },
+        async getChannels() {
+            // TODO fix me, I'm hardcoded and have 1 option
+            return [{ value: "channel_a", label: "Channel A" }];
+        },
+        async getFactories() {
+            // TODO fix me, I'm hardcoded and have 5 option
+            return [
+                { value: "factory_a", label: "Factory A" },
+                { value: "factory_b", label: "Factory B" },
+                { value: "factory_c", label: "Factory C" },
+                { value: "factory_d", label: "Factory D" },
+                { value: "factory_e", label: "Factory E" }
+            ];
+        },
+        removeInvalidChoices() {

```

Comment:
Can you find a better name? They're not actually invalid, they just have <=1 options.